StripePaymentSheet/StripePaymentSheet/PaymentSheet/Intent.swift

```diff
@@ -199,18 +201,28 @@ extension STPConfirmPaymentMethodOptions {
         paymentMethodType: STPPaymentMethodType,
         customer: PaymentSheet.CustomerConfiguration?
     ) {
-        // This property cannot be set if there is no customer.
+        // Something went wrong if we're trying to save and there's no Customer!
         assert(!(shouldSave && customer == nil))
 
-        // Only support card and US bank setup_future_usage in payment_method_options
-        guard customer != nil && paymentMethodType == .card || paymentMethodType == .USBankAccount
-        else {
+        // This property should only be set if
+        // 1. We're displaying a "Save this pm for future payments" checkbox
+        // 2. The PM type is card or US bank
+        guard customer != nil && paymentMethodType == .card || paymentMethodType == .USBankAccount else {
+            return
+        }
+        // Passing "none" *overrides* the top-level setup_future_usage and is not what we want, b/c this code is called even when we don't display the "save" checkbox (e.g. when the PI top-level setup_future_usage is already set).
+        // Instead, we pass an empty string to 'unset' this value. This makes the PaymentIntent *inherit* the top-level setup_future_usage.
+        let sfuValue = shouldSave ? "off_session" : ""
+        switch paymentMethodType {
+        case .card:
+            cardOptions = cardOptions ?? STPConfirmCardOptions()
+            cardOptions?.additionalAPIParameters["setup_future_usage"] = sfuValue
+        case .USBankAccount:
+            usBankAccountOptions = usBankAccountOptions ?? STPConfirmUSBankAccountOptions(setupFutureUsage: .none)
```

Comment:
It does, but `additionalAPIParameters` overwrites it. IMO this is by design and we should be able to rely on this overwriting, since `additionalAPIParameters` property is an escape hatch.

Reply:
This is a bit smelly IMO, especially because you are explicitly creating the instance with one value and then you are immediately potentially overriding it.
Not saying is worth doing a big refactor over it, but we should at least clearly document what's going on here, i.e., the SFU value passed when creating the instance will effectively be ignored and is only used because it is required by the API.

Reply:
Agree it's smelly and ideally shouldn't be necessary. I think we can fix this as a follow-up to https://jira.corp.stripe.com/browse/RUN_MOBILESDK-1737 - I've attempted to describe this in the description.

For now, I'll add some documentation as suggested